README.md

```diff
@@ -1,23 +1,1377 @@
+  plug :fetch_flash
+  plug :protect_from_forgery
+  plug :put_secure_browser_headers
+end
+```
+
+Replace the line `plug :fetch_flash`
+with `plug :fetch_live_flash`
+such that it now looks like this:
+
+```elixir
+pipeline :browser do
+  plug :accepts, ["html"]
+  plug :fetch_session
+  plug :fetch_live_flash
+  plug :protect_from_forgery
+  plug :put_secure_browser_headers
+end
+```
+
+That ensures flash messages (_e.g: "not connected to network"_)
+are displayed in the client when the LiveView App is running.
+
+> 🏁 At the end of Step 4, the
+[`lib/live_view_counter_web/router.ex`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3b562348100ce0048da3f1f4e81c036d94e6463e/lib/live_view_counter_web/router.ex#L8)
+should look like:
+[`router.ex#L8`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3b562348100ce0048da3f1f4e81c036d94e6463e/lib/live_view_counter_web/router.ex#L8)
+
+<br />
+
+### Step 5: Add `Phoenix.LiveView` Helpers to `live_view_counter_web.ex`
+
+
+Open the
+[`lib/live_view_counter_web.ex`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/7e300eadb4b71443543d33dc9e02975a99f0aa08/lib/live_view_counter_web.ex)
+file
+and add the relevant `Phoenix.LiveView` import statements
+for each of the `controller`, `view` and `router` blocks.
+
+```diff
+def controller do
+  quote do
+    ...
++   import Phoenix.LiveView.Controller
+  end
+end
+
+def view do
+  quote do
+    ...
++   import Phoenix.LiveView.Helpers
+  end
+end
+
+def router do
+  quote do
+    ...
++   import Phoenix.LiveView.Router
+  end
+end
+```
+
+> 🏁 Changes made in Step 5: <br />
+**Before**:
+[`lib/live_view_counter_web.ex`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/7e300eadb4b71443543d33dc9e02975a99f0aa08/lib/live_view_counter_web.ex) <br />
+***After***:
+[`lib/live_view_counter_web.ex#L27`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3c7c448ead7c161167fb310638b44be80b20ea1e/lib/live_view_counter_web.ex#L27)
+The relevant lines are
+[`27`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3c7c448ead7c161167fb310638b44be80b20ea1e/lib/live_view_counter_web.ex#L27),
+[`46`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3c7c448ead7c161167fb310638b44be80b20ea1e/lib/live_view_counter_web.ex#L46)
+and
+[`55`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3c7c448ead7c161167fb310638b44be80b20ea1e/lib/live_view_counter_web.ex#L55).
+
+<br />
+
+### Step 6: Create the `/live` Socket in `endpoint.ex`
+
+In order to allow the client(s)
+to communicate with the server,
+we need to open a socket.
+Open the
+[`lib/live_view_counter_web/endpoint.ex`](https://github.com/dwyl/phoenix-liveview-counter-tutorial/blob/3c7c448ead7c161167fb310638b44be80b20ea1e/lib/live_view_counter_web/endpoint.ex)
+file
+and add the following lines to it:
+
+```elixir
+socket "/live", Phoenix.LiveView.Socket,
+  websocket: [connect_info: [session: @session_options]]
```

Comment:
Do we also need to indicate how to define the `@session_options`? It wasn't already defined for me (not sure if it is supposed to be added when the live view dependency is added)

Reply:
Oh, fair enough. When I created a new phoenix app (Step 1) and installed `phoenix_live_view` (Step 2) a lot of the setup was performed automagically ... I wonder if it's a Mac thing? 🤷‍♂ 

It appears we need to add a bit more from the official installation instructions:
https://hexdocs.pm/phoenix_live_view/installation.html

![image](https://user-images.githubusercontent.com/194400/76306592-1a6f5180-62bf-11ea-9f86-33197448a044.png)

I can do this now.

Reply:
Section added: https://github.com/dwyl/phoenix-liveview-counter-tutorial/pull/4/commits/8b7fb02ea1587dd215ec59748a23b83a32a57336

Reply:
I've updated my Phoenix to 1.4.15 and the `@session_options` is now defined when I installed the dependency

Reply:
@SimonLab yeah, `1.4.15` appears to add some enhancements to post install. ✨ 
I think the aim of the Phoenix team is reduce the number of configuration steps to using LiveView.
With LiveView, Phoenix has the potential to _replace_ React (_or other front-end frameworks_) so they want/need to make the setup process as fast as possible. 💡